worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,211 @@
+                                                        'World 2': lambda state: state.has('World 2 Gate', player),
+                                                        'World 3': lambda state: state.has('World 3 Gate', player),
+                                                        'World 4': lambda state: state.has('World 4 Gate', player),
+                                                        'World 5': lambda state: state.has('World 5 Gate', player),
+                                                        'World 6': lambda state: state.has('World 6 Gate', player)})
+    CurWorld = 1
+    CurLev = 1
+    for i in range(47):
+        multiworld.get_region(f'World {CurWorld}', player).add_exits([gamevar.level_location_list[i]])
+        if CurLev > 7:
+            CurLev = 1
+            CurWorld += 1
+        else: CurLev += 1
+
+    multiworld.get_region('1-4', player).add_exits([gamevar.boss_order[0]],{gamevar.boss_order[0]: lambda state: logic._14Clear(state)})
+    multiworld.get_region('1-8', player).add_exits([gamevar.boss_order[1]],{gamevar.boss_order[1]: lambda state: logic._18Clear(state)})
+    multiworld.get_region('2-4', player).add_exits([gamevar.boss_order[2]],{gamevar.boss_order[2]: lambda state: logic._24Clear(state)})
+    multiworld.get_region('2-8', player).add_exits([gamevar.boss_order[3]],{gamevar.boss_order[3]: lambda state: logic._28Clear(state)})
+    multiworld.get_region('3-4', player).add_exits([gamevar.boss_order[4]],{gamevar.boss_order[4]: lambda state: logic._34Clear(state)})
+    multiworld.get_region('3-8', player).add_exits([gamevar.boss_order[5]],{gamevar.boss_order[5]: lambda state: logic._38Clear(state)})
+    multiworld.get_region('4-4', player).add_exits([gamevar.boss_order[6]],{gamevar.boss_order[6]: lambda state: logic._44Clear(state)})
+    multiworld.get_region('4-8', player).add_exits([gamevar.boss_order[7]],{gamevar.boss_order[7]: lambda state: logic._48Clear(state)})
+    multiworld.get_region('5-4', player).add_exits([gamevar.boss_order[8]],{gamevar.boss_order[8]: lambda state: logic._54Clear(state)})
+    multiworld.get_region('5-8', player).add_exits([gamevar.boss_order[9]],{gamevar.boss_order[9]: lambda state: logic._58Clear(state)})
+    multiworld.get_region('World 6', player).add_exits(["6-8"],{"6-8": lambda state: bosses.castle_access(state)})
+    multiworld.get_region('6-4', player).add_exits([gamevar.boss_order[10]],{gamevar.boss_order[10]: lambda state: logic._64Clear(state)})
+    multiworld.get_region('6-8', player).add_exits(["Bowser's Room"],{"Bowser's Room": lambda state: bosses.castle_clear(state)})
+
+    if get_option_value(multiworld, player, "extras_enabled") == 1:
+        for i in range(6):
+            multiworld.get_region(f'World {i + 1}', player).add_exits([f'{i + 1}-Extra'],
+                                                                {f'{i+1}-Extra': lambda state: state.has('Extra Panels', player) or state.has(f'Extra {i + 1}', player)})

```

Comment:
This line is definitely too long off the right, and also not formatted correctly. Again, I'd suggest setting up automatic code style checking for your IDE.